voluptuous/schema_builder.py

```diff
@@ -216,15 +221,11 @@ def _compile_mapping(self, schema, invalid_msg=None):
         invalid_msg = invalid_msg or 'mapping value'
 
         # Keys that may be required
-        all_required_keys = set(key for key in schema
-                                if key is not Extra
-                                and ((self.required and not isinstance(key, (Optional, Remove)))
-                                     or isinstance(key, Required)))
+        all_required_keys = set(key for key in schema if key is not Extra and (
+            (self.required and not isinstance(key, (Optional, Remove))) or isinstance(key, Required)))
```

Comment:
I think this (and the below) are less clear. The underlying issue is just about operators starting a line, which can be simply fixed by putting just the operators at the end of the preceding line.